test/integration/features/profile.feature

```diff
@@ -2,104 +2,251 @@
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+      When executing "minishift config set memory 4096" succeeds
+       And executing "minishift config set disk-size 30g" succeeds
+       And executing "minishift config set cpus 4" succeeds
+       And executing "minishift addons enable registry-route" succeeds
+         
+  @minishift-only
+  Scenario: As user, I can check the config settings and add-ons list that should not be changed in profile 'minishift' 
+    Given Profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+      When executing "minishift --profile minishift config get memory" succeeds
+       Then stdout should match "<nil>"
+      When executing "minishift --profile minishift config get disk-size" succeeds
+       Then stdout should match "<nil>"
+      When executing "minishift --profile minishift config get cpus" succeeds
+       Then stdout should match "<nil>"
+      When executing "minishift --profile minishift addons list"
+       Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+       And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+       And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+       And stdout should match "xpaas\s*: disabled\s*P\(0\)"
 
-      """
-    When executing "minishift profile set minishift" succeeds
+  Scenario: As a user, I can start profile 'foo'
+    Given Profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift start" succeeds
       Then stdout should contain
       """
-      Profile 'minishift' set as active profile.
+      Add-on 'registry-route' created docker-registry route.
       """
-Scenario: User should not be able to create profile with blank profile name
-    When executing "minishift profile set"
-     Then exitcode should equal "1"
-     Then stderr should contain
+      And Profile "foo" should have state "Running"
+  
+  @minishift-only
+  Scenario: As user, I can list add-ons in profile 'foo' 
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+      And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+      And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+      And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: Checking that memory value was applied on profile 'foo'
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
```

Comment:
So why do you execute this all the time?

Reply:
Each scenario should be satisfied the precondition and less depend on the previous scenario. 
For example let foo remains an active profile at the end of the scenario execution. For the very next scenario as a user  i am not aware of the current active profile though the profile context is transferred form the previous step. So to make the steps more descriptive each time i check this as a precondition when required.

Reply:
However in this case this should not be allowed as this is not a feature of profile. It just a check on config applied on running VM. Moreover this is completely depends on the previous scenario, adding this as a scenario is to avoid the step skip if any failure occurs on any config applied.